src/act_wiz.cpp

```diff
@@ -1767,10 +1767,7 @@ void do_switch(CHAR_DATA *ch, const char *argument) {
         return;
     }
 
-    ch->desc->character = victim;
-    ch->desc->original = ch;
-    victim->desc = ch->desc;
-    ch->desc = nullptr;
+    ch->desc->do_switch(victim);
     /* change communications to match */
     victim->comm = ch->comm;
```

Comment:
I'm wondering whether it's rolling these last 3 lines into do_switch().

Reply:
Yeah I wondered about that. It's a lot less about the descriptor and more about the CHAR_DATA though. For now I left them, thinking the descriptor is (mostly) about the lower-level stuff....though I can be swayed.

There's a lot of stuff in CHAR_DATA that should probably just be in descriptor.

I realise now the victim's `comm` is left forever in whatever state the last switcher had.